GliaWidgets/Sources/Download/FileDownload.Environment.Interface.swift

```diff
@@ -1,9 +1,43 @@
 extension FileDownload {
     struct Environment {
-        var fetchFile: CoreSdkClient.FetchFile
+        var fetchFile: FetchFile
         var fileManager: FoundationBased.FileManager
         var gcd: GCD
         var localFileThumbnailQueue: FoundationBased.OperationQueue
         var uiImage: UIKitBased.UIImage
     }
 }
+
+extension FileDownload.Environment {
+    enum FetchFile {
+        case fromEngagement(CoreSdkClient.FetchFile)
+        case fromSecureMessaging(CoreSdkClient.DownloadSecureFile)
+    }
+}
+extension FileDownload.Environment.FetchFile {
+    @discardableResult
+    func fetchFile(
+        _ file: CoreSdkClient.EngagementFile,
+        progress: @escaping CoreSdkClient.EngagementFileProgressBlock,
+        completion: @escaping (Result<CoreSdkClient.EngagementFileData, Error>) -> Void) -> CoreSdkClient.Salemove.Cancellable? {
+            switch self {
+            case let .fromSecureMessaging(fetch):
+                return fetch(file, progress, completion)
+            case let .fromEngagement(fetch):
+                fetch(file, progress) { data, error in
+                    switch (data, error) {
+                    case (.none, .none):
+                        break
+                    case let (.some, .some(error)):
+                        completion(.failure(error))
+                    case let (.some(fileData), .none):
+                        completion(.success(fileData))
+                    case let (.none, .some(error)):
+                        completion(.failure(error))
```

Comment:
could we simplify and union two cases?
```suggestion
                    case let (_, .some(error)):
                        completion(.failure(error))
                    case let (.some(fileData), .none):
                        completion(.success(fileData))
```